libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1161,25 +1206,53 @@ void WaveClip::ShiftBy(double delta) noexcept
 
 // Bug 2288 allowed overlapping clips.
 // This was a classic fencepost error.
-// We are within the clip if start < t <= end.
+// We are within the clip if start <= t < end.
 // Note that BeforeClip and AfterClip must be consistent
 // with this definition.
 bool WaveClip::WithinPlayRegion(double t) const
 {
-    auto ts = TimeToSamples(t);
-    return ts > GetPlayStartSample() && ts < GetPlayEndSample() + GetAppendBufferLen();
+   return GetPlayStartTime() <= t && t < GetPlayEndTime();
+}
+
+bool WaveClip::EntirelyWithinPlayRegion(double t0, double t1) const
+{
+   assert(t0 <= t1);
+   return !BeforePlayRegion(t0) && !AfterPlayRegion(t1);
+}
+
+bool WaveClip::PartlyWithinPlayRegion(double t0, double t1) const
+{
+   assert(t0 <= t1);
+   return OverlapsPlayRegion(t0, t1) && !EntirelyWithinPlayRegion(t0, t1);
+}
+
+bool WaveClip::OverlapsPlayRegion(double t0, double t1) const
+{
+   assert(t0 <= t1);
+   return !(BeforePlayRegion(t1) || AfterPlayRegion(t0));
+}
+
+bool WaveClip::ExtendsPlayRegion(double t0, double t1) const
+{
```

Comment:
This one is the test for what you might describe as "properly partial overlap" excluding the case that either interval is a subset of the other